lmp/model/_transformer.py

```diff
@@ -0,0 +1,199 @@
+import math
+import torch
+import numpy as np
+from torch.utils.tensorboard import SummaryWriter
+from torch.nn import Linear
+from torch.nn.functional import softmax, relu
+
+
+class PositionalEncoding(torch.nn.Module):
+    "Implement the PE function."
+
+    def __init__(self, d_model, dropout, max_len=5000):
+        super(PositionalEncoding, self).__init__()
+        self.dropout = torch.nn.Dropout(p=dropout)
+
+        # Compute the positional encodings once in log space.
+        pe = torch.zeros(max_len, d_model)
+        position = torch.arange(0, max_len).unsqueeze(1)
+        div_term = torch.exp(torch.arange(0, d_model, 2) *
+                             -(math.log(10000.0) / d_model))
+        pe[:, 0::2] = torch.sin(position * div_term)
+        pe[:, 1::2] = torch.cos(position * div_term)
+        pe = pe.unsqueeze(0)
+        self.register_buffer('pe', pe)
+
+    def forward(self, x):
+        x = x + torch.autograd.Variable(self.pe[:, :x.size(-2)],
```

Comment:
remove old version autograd code (no need to use `Variable`, simply use `self.pe[:, :x.size(-2)]`).